shell/platform/android/test/io/flutter/embedding/engine/renderer/FlutterRendererTest.java

```diff
@@ -286,6 +284,7 @@ public void itConvertsDisplayFeatureArrayToPrimitiveArrays() {
             new Rect(50, 60, 70, 80), FlutterRenderer.DisplayFeatureType.CUTOUT));
 
     // Execute the behavior under test.
+    clearInvocations(fakeFlutterJNI);
```

Comment:
What changed to make a call to `clearInvocations` necessary, but just in this test? (the docs say to [avoid this method at all costs](https://javadoc.io/doc/org.mockito/mockito-core/2.25.1/org/mockito/Mockito.html#clearInvocations-T...-) so I just want to know why we need it here)

Reply:
That's a good question.

The tl;dr is because calls to setViewportMetrics happen ambiently in the background with the new setup. I could change this back to creating the renderer manually (not created via engine) and avoid the need for this call. Wdut?

Reply:
I talked to Gray offline and decided it would be more work to document why this is OK than just not use it.

Resolved!